src/schema/decomposition/commands/clone-decomposition-from-previous-survey.handler.ts

```
@@ -10,10 +11,12 @@ import { CloneDecompositionFromPreviousSurveyCommand } from './clone-decompositi
 export class CloneDecompositionFromPreviousSurveyHandler
 	implements ICommandHandler<CloneDecompositionFromPreviousSurveyCommand>
 {
-	constructor(private decompositionRepository: DecompositionRepository) {}
+	constructor(private decompositionRepository: DecompositionRepository, private surveyRepository: SurveyRepository) {}
 
 	public async execute(command: CloneDecompositionFromPreviousSurveyCommand): Promise<Element[]> {
-		const previousSurveyId = await this.decompositionRepository.findPreviousSurveyId(command.surveyId);
+		const previousSurveyId = await this.surveyRepository.findIdPreviousSurveyWithNen2767Decomposition(
```

Shouldn't a survey service be asked to provide us with this instead of going straight to the repository?

---

I do notice that to be a pattern in the codebase indeed. Will happily change for the sake of code consistency, but what is the benefit of doing so versus directly asking the repository?

From what I understand a repository is an abstraction that specifically serves the purpose of communicating with a data source (in this case prisma -> sql -> postgres). Correct me if I'm wrong, but that's the only thing I'm looking for here.

The functional difference I do spot in places where this Service/Factory/Repository pattern is in place is that sometimes there's logic in the Factory that enriches the data received directly from prisma. I do certainly see a Service having value there (as the glue between repository and factory).

---

It's a bit of an arbitrary thing I believe. I'm happy to go straight to the repository until a service makes sense to keep business logic out of a controller/resolver. I suspect we have another team member who might have a different opinion ;) You can also see in some places we don't have a purely pass-through service in between but go straight to the repository which I think is fine. As soon as a custom exception needs throwing though which gets exposed via GQL, I'd strongly consider doing that from a service and not the repository. I was curious about your POV and am happy to agree we don't make it a hard rule.

---

Alright, good to discuss then. Good point regarding exceptions: I do have a custom exception thrown from the repository here and I did notice some odd behaviour when doing tests locally breaking the (Node) process. Will move the throwing of the Exception to the Service layer and hopefully that fixes it, so I suppose that's a good reason to apply your initial suggesting of putting a service in between. 

BTW: Is there some sort of rule of thumb regarding throwing Exceptions in Node/NestJS that I'm not aware of? I do find some SO threads advising to generally throw Exceptions from Services (instead of from a Controller -- but I guess that applies to Repositories as well) but I can't seem to find a reason why.